### PR TITLE
Add payment support

### PR DESCRIPTION
**Pull request by asood123**

### CODE REVIEW

File: server/graphql/mutations.js [hunk 1]
```diff
@@ -69,6 +76,26 @@ const mutations = {
         status: response.status,
         description: response.description
       }))
+
+      // record payment, if needed
+      .then(responseModel => {
+        if (response.user.card && response.user.card.token && tier.amount > 0) {
```
Comment (xdamman) [inline server/graphql/mutations.js]: if needed should be defined by amount, not by whether there is a user.card.
And we should return a proper error message if there isn't. Not clear by reading this code what are the possible error cases and messages

File: server/lib/activities.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ export default {
     // get subscription data
     if (activity.data.subscription) {
       ({ interval } = activity.data.subscription);
+      amount = amount || activity.data.subscription.amount/100;
```
Comment (xdamman) [inline server/lib/activities.js]: we should divide by 100 at the very last moment when we display a string.

File: server/models/Tier.js [hunk 3]
```diff
@@ -88,13 +106,13 @@ export default function(Sequelize, DataTypes) {
             } else if (this.maxQuantity) {
               return this.maxQuantity;
             } else {
-              return -1; // means there was no maxQuantity set, so infinite availability
+              return Infinity;
```
Comment (xdamman) [inline server/models/Tier.js]: What about keeping a constant somewhere throughout? e.g. `constants.INFINITY`?

File: templates/emails/ticket.confirmed.hbs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+{{> toplogo}}
+
+{{#if user.name}}
+<p>Dear {{user.name}},</p>
+{{else}}
+<p>Hi!</p>
+{{/if}}
+
+<p> We are excited that you'll be joining us at {{ event.name }} on {{ event.startsAt }} at {{ event.locationName }} </p>
```
Comment (xdamman) [inline templates/emails/ticket.confirmed.hbs]: should remove the space after `<p>`, also below

File: templates/emails/ticket.confirmed.hbs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+<p> This confirms your {{ response.quantity }} {{{pluralize "ticket" n=response.quantity}}}. </p>
+
+<p>In case you didn't know, we manage our budget transparently. You can follow our future activities and how we will be spending this money on our open collective page:
+  <a href="https://opencollective.com/{{group.slug}}">https://opencollective.com/{{group.slug}}</a>.
```
Comment (xdamman) [inline templates/emails/ticket.confirmed.hbs]: add utm: `<a href="https://opencollective.com/{{group.slug}}?{{utm}}">`

File: templates/emails/ticket.confirmed.hbs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+{{> footer}}
+
+<p style="text-align:center;font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;box-sizing:border-box;font-size:12px;color:#999;">
+OpenCollective, Inc.<br />
+EIN: 47-3896707<br />
+340 S LEMON AVE #3717<br />
+Walnut CA 91789<br />
+USA
+</p>
```
Comment (xdamman) [inline templates/emails/ticket.confirmed.hbs]: I'd remove this.

File: test/lib.createPayment.test.js [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+              token: STRIPE_TOKEN,
+              amount: AMOUNT,
+              currency: CURRENCY
+            }
+          }))
+          .catch(err => expect(err.message).to.equal('The host for the collective slug wwcode-austin has no Stripe account set up'))
+
+        });
+
+        it('if stipe has live key and not in production', () => {
```
Comment (xdamman) [inline test/lib.createPayment.test.js]: stripe

File: server/lib/activities.js (hunk 2 above)
Comment (asood123) [inline server/lib/activities.js]: we convert it into string in the next line :)

File: server/models/Tier.js (hunk 3 above)
Comment (asood123) [inline server/models/Tier.js]: Infinity is a supported JS concept. Only place it's used is in Graphql types right now. If needed elsewhere, we can make it a constant.

File: server/graphql/mutations.js (hunk 1 above)
Comment (asood123) [inline server/graphql/mutations.js]: ah yes, i'll add an error message.

File: templates/emails/ticket.confirmed.hbs (hunk 6 above)
Comment (asood123) [inline templates/emails/ticket.confirmed.hbs]: done

File: test/lib.createPayment.test.js (hunk 7 above)
Comment (asood123) [inline test/lib.createPayment.test.js]: done

File: templates/emails/ticket.confirmed.hbs (hunk 5 above)
Comment (asood123) [inline templates/emails/ticket.confirmed.hbs]: added

File: templates/emails/ticket.confirmed.hbs (hunk 4 above)
Comment (asood123) [inline templates/emails/ticket.confirmed.hbs]: done

**Pull request merged**